src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -134,17 +159,39 @@ void HSHomeObject::init_homestore() {
         RELEASE_ASSERT(!_our_id.is_nil(), "Received no SvcId and need FORMAT!");
         LOGW("We are starting for the first time on [{}], Formatting!!", to_string(_our_id));
 
-        HomeStore::instance()->format_and_start({
-            {HS_SERVICE::META, hs_format_params{.size_pct = 5.0}},
-            {HS_SERVICE::LOG, hs_format_params{.size_pct = 10.0, .chunk_size = 32 * Mi}},
-            {HS_SERVICE::REPLICATION,
-             hs_format_params{.size_pct = 79.0,
-                              .num_chunks = 65000,
-                              .block_size = _data_block_size,
-                              .alloc_type = blk_allocator_type_t::append,
-                              .chunk_sel_type = chunk_selector_type_t::CUSTOM}},
-            {HS_SERVICE::INDEX, hs_format_params{.size_pct = 5.0}},
-        });
+        if (has_data_dev && has_fast_dev) {
+            // Hybrid mode
+            LOGD("Has both Data and Fast, running with Hybrid mode");
+            HomeStore::instance()->format_and_start({
+                {HS_SERVICE::META, hs_format_params{.dev_type = HSDevType::Fast, .size_pct = 9.0, .num_chunks = 64}},
+                {HS_SERVICE::LOG,
+                 hs_format_params{.dev_type = HSDevType::Fast, .size_pct = 45.0, .chunk_size = 32 * Mi}},
+                {HS_SERVICE::INDEX, hs_format_params{.dev_type = HSDevType::Fast, .size_pct = 45.0, .num_chunks = 128}},
+                {HS_SERVICE::REPLICATION,
+                 hs_format_params{.dev_type = HSDevType::Data,
+                                  .size_pct = 99.0,
+                                  .num_chunks = 60000,
+                                  .block_size = _data_block_size,
+                                  .alloc_type = blk_allocator_type_t::append,
+                                  .chunk_sel_type = chunk_selector_type_t::CUSTOM}},
+            });
+        } else {
+            auto run_on_type = has_fast_dev ? homestore::HSDevType::Fast : homestore::HSDevType::Data;
+            LOGD("Running with Single mode, all service on {}", run_on_type);
+            HomeStore::instance()->format_and_start({
+	        // FIXME:  this is to work around the issue in HS that varsize allocator doesnt work with small chunk size.
+                {HS_SERVICE::META, hs_format_params{.dev_type = run_on_type, .size_pct = 5.0, .num_chunks = 1}},
+                {HS_SERVICE::LOG, hs_format_params{.dev_type = run_on_type, .size_pct = 10.0, .chunk_size = 32 * Mi}},
+                {HS_SERVICE::INDEX, hs_format_params{.dev_type = run_on_type, .size_pct = 5.0, .num_chunks = 1}},
+                {HS_SERVICE::REPLICATION,
+                 hs_format_params{.dev_type = run_on_type,
+                                  .size_pct = 79.0,
+                                  .num_chunks = 65000,
```

Comment:
this should also be adjusted, right? 10 pct of total device would be still very large and exceed 64K in total? 

Also can we be more conservative say setting this to 40000, before the fixme part is done?

Reply:
yes, lets postpone this calculation and will get it from testing on real environment.